Lib/unittest/mock.py

```diff
@@ -402,6 +403,10 @@ def __init__(self, /, *args, **kwargs):
 class NonCallableMock(Base):
     """A non-callable version of `Mock`"""
 
+    # Store a mutex as a class attribute to protect concurrent accesses to
+    # a NonCallableMock's other attributes.
+    _lock = RLock()
```

Comment:
Worth double-checking with the coding standards but I would personally find it easier to see this defined right before its user, i.e. `__getattr__`.

Reply:
I'm not sure about the coding standard either; I just usually see class attributes declared at the beginning of a class, before `__init__()`. Your reasoning also makes sense though. I think I'll leave it for now but also leave this discussion open in case there are more opinions.